[ritul] iP

ritulkrsingh opened the pull request:
# BRMO - BetaRMinusOne

BRMO is a task management application designed to help you keep track of your tasks efficiently.

## Features

- Add tasks
- Remove tasks (!!!)
- Mark tasks as done
- Mark tasks as undone (🤯)
- Find tasks by keyword
- Save and load tasks from a file

## Getting Started

1. Clone the repository:
   ```sh
   git clone https://github.com/yourusername/BRMO.git
   ```
2. Navigate to the project directory:
   ```sh
   cd BRMO
   ```
3. Build the project:
   ```sh
   ./gradlew build
   ```
4. Run the application:
   ```sh
   ./gradlew run
   ```

## To-Do List

- [x] Implement task addition
- [x] Implement task removal
- [ ] Implement task editing
- [ ] Add more unit tests

> "How does a project get to be a year late? One day at a time." - *someone, probably*

Check out the (non-existent) [documentation](https://github.com/yourusername/BRMO/wiki) for more details.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ritulkrsingh committed:
Named the bot.

ritulkrsingh committed:
Added echo back functionality.

ritulkrsingh committed:
Added list functionality.

ritulkrsingh committed:
Add mark/unmark option

ritulkrsingh committed:
Add deadlines/events/todos

ritulkrsingh committed:
Support testing

ritulkrsingh committed:
Add InvalidCommandException

ritulkrsingh committed:
Support deletion of tasks

ritulkrsingh committed:
Add data file support

ritulkrsingh committed:
Merge branch 'branch-Level-7'
Merge branch-Level-7 to master

ritulkrsingh committed:
Add date formats

ritulkrsingh committed:
Merge branch 'branch-Level-8'

Make and merge branch-Level-8

testing1234567891011121314 reviewed (commented):
LGTM.

testing1234567891011121314 commented on src/main/java/BRMO.java:
```diff
@@ -0,0 +1,133 @@
+                String line = fileScanner.nextLine();
+                String[] split = line.split(" \\| ");
+
+                switch (split[0]) {
+                case "T":
+                    tasks.add(new Todo(split[2]));
+                    break;
+                case "D":
+                    tasks.add(new Deadline(split[2], split[3]));
+                    break;
+                case "E":
+                    tasks.add(new Event(split[2], split[3], split[4]));
+                    break;
+                }
```
Good job on the indentation for the switch case!

testing1234567891011121314 commented on src/main/java/BRMO.java:
```diff
@@ -0,0 +1,133 @@
+    public static void main(String[] args) throws InvalidCommandException {
+        System.out.println("Hello! I'm BetaRMinusOne; a.k.a. BRMO.");
+
+        ArrayList<Task> tasks = new ArrayList<Task>();
```
Good use of names!

testing1234567891011121314 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.Date;
+
+public class Event extends Task {
+    private Date timeL;
+    private Date timeR;
```
Good use of camelcase for naming variables, possible suggestion to use variable names timeStart/timeEnd for more information?

testing1234567891011121314 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + outputFormat.format(timeL) + " to: " + outputFormat.format(timeR) + ")";
```
Hi, can possibly consider use of a new line to break this return statement into two if it is more than character limit?

testing1234567891011121314 reviewed (commented):
LGTM, minor nits.

ritulkrsingh committed:
Add more OOP

ritulkrsingh committed:
Move everything to package

ritulkrsingh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Add gradle support to automate project builds

ritulkrsingh committed:
Merge and set up Gradle

ritulkrsingh committed:
Add JUnit testing and create JAR package

ritulkrsingh committed:
Add javadoc headers for classes and methods

ritulkrsingh committed:
Fix coding standard issues

ritulkrsingh committed:
Add find feature

ritulkrsingh committed:
Merge branch 'branch-A-CodingStandard'
Merge the parallel branches

ritulkrsingh committed:
Merge branch 'branch-A-JavaDoc'

Merge another branch

ritulkrsingh committed:
Merge branch 'branch-Level-9'

Merge Level-9 to master.

ritulkrsingh committed:
Fix incorrectly loading marked tasks from file

ritulkrsingh committed:
Create a GUI

ritulkrsingh committed:
Fix text-boxes resizing when too many messages

ritulkrsingh committed:
Add javadoc to Ui.java

ritulkrsingh committed:
Add GUI with JavaFX

ritulkrsingh committed:
Merge branch 'branch-Level-10'
Merge GUI branch

ritulkrsingh committed:
Add assertions

Several methods in the code could benefit from assertions.

For example whenever a message confirming (un)marking of a task
is sent, it can be asserted that the task is in fact (un)marked.

ritulkrsingh committed:
Improve code quality

The variable name "split" has been avoided to avoid confusion with a
String method of the same name.

Magic numbers were used in Parser.java, which have been replaced to
better describe what's going on.

A reduntant re-implementation of toString() for TaskList has been fixed.

ritulkrsingh committed:
Merge pull request #2 from ritulkrsingh/branch-A-Assertions

Add assertions

ritulkrsingh committed:
Merge branch 'branch-A-CodeQuality' into master

ritulkrsingh committed:
Add user guide

ritulkrsingh committed:
Fix compile error

ritulkrsingh committed:
Add welcome message

ritulkrsingh committed:
Make Launcher class

ritulkrsingh committed:
Add anchors to GUI elements

ritulkrsingh committed:
Set JAR window title to BRMO
